src/dsp_tools/utils/xml_upload.py

```diff
@@ -475,10 +475,12 @@ def xml_upload(
     preparation_start = datetime.now()
 
     # Connect to the DaSCH Service Platform API and get the project context
-    con = Connection(server)
-    try_network_action(failure_msg="Unable to login to DSP server", action=lambda: con.login(user, password))
-    proj_context = try_network_action(failure_msg="Unable to retrieve project context from DSP server",
-                                      action=lambda: ProjectContext(con=con))
+    con = login(server=server, user=user, password=password)
+    try:
+        proj_context = try_network_action(lambda: ProjectContext(con=con))
+    except BaseError:
+        logger.error("Unable to retrieve project context from DSP server", exc_info=True)
+        raise UserError("Unable to retrieve project context from DSP server") from None
```

Comment:
Until now, the error "Unable to retrieve project context from DSP server" escalated uncontrolled. Now, it is logged incl. stacktrace, and a user-friendly UserError is raised instead